lib/SILGen/SILGenMaterializeForSet.cpp

```diff
@@ -964,7 +964,7 @@ MaterializeForSetEmitter::createSetterCallback(SILFunction &F,
     }
 
     // The callback gets the address of 'self' at +0.
-    ManagedValue mSelf = ManagedValue::forLValue(self);
+    ManagedValue mSelf = ManagedValue::forUnmanaged(self);
```

Comment:
I wonder if there is a more explicit API we could use depending on whether ``self`` is ``@inout`` or ``@in_guaranteed``: ``forLValue`` and ``forBorrowedAddressRValue`` maybe (though we can have trivial r-values).
Not that this should block this commit.